base/compiler/typelattice.jl

```diff
@@ -122,6 +122,8 @@ function ⊑(@nospecialize(a), @nospecialize(b))
     (a === Any || b === NOT_FOUND) && return false
     a === Union{} && return true
     b === Union{} && return false
+    @assert !isa(a, TypeVar) || a
+    @assert !isa(b, TypeVar) || b
```

Comment:
we use this pattern to print the exact bad `TypeVar` ?

Reply:
With `@test` that works pretty well. But I'm going to remove this, since here it just throws `TypeError: non-boolean (TypeVar) used in boolean context`, which isn't helpful.